### DISCUSSION TITLE
Question about `generate_spatial_bounding_box`

### DISCUSSION THREAD

**Discussion by yiheng-wang-nv** [Project-MONAI/MONAI#3471] [Ideas]
Hi, I just found that the `generate_spatial_bounding_box` may create  invalid coordinates.
For example:

```
a = torch.zeros([1, 100, 100])
a[0, -1, -1] = 1
```
The output of `generate_spatial_bounding_box(a)` is:
`([99, 99], [100, 100])`

However, the valid coordinates of `a` is 0 to 99, and cannot be 100.

I think This line:
https://github.com/Project-MONAI/MONAI/blob/89f0bd488851bfb012541ad6344cf0fbba9f9431/monai/transforms/utils.py#L916
may be we should not do `+1`.

Am I right?

**Comment by yiheng-wang-nv** [comment]
Hi @Nic-Ma , @wyli , what do you think?

**Reply by Nic-Ma** [reply]
Hi @yiheng-wang-nv ,

Thanks for your finding, I am not whether it's a bug, let me try to double-check ASAP.

**Comment by wyli** [comment]
I think that's the intended behaviour, the function computes the coordinates, and then we can index the array with `a[:, 99:100, 99:100]` (it returns `tensor([[[1.]]])`)

**Reply by Nic-Ma** [reply]
Hi @yiheng-wang-nv ,

I enhanced the doc-string for this question in another transform doc related PR: https://github.com/Project-MONAI/MONAI/pull/3494.

Thanks.
